### DISCUSSION TITLE
Using DiffPool on a large graph

### DISCUSSION THREAD

**Discussion by mdanb** [pyg-team/pytorch_geometric#2518] [Q&A] [answered]
I have a large graph (~20 000 nodes, 5 million edges) and I'd like to use DiffPool. Is there any way to make it memory efficient? The only way I can think of is to use some predetermined clustering algorithm on the initial graph to make it smaller, and then run DiffPool on that but I'm not sure that's the best idea

**Comment by zcaicaros** [comment, marked as answer]
The reason why DiffPool consumes much memory is because it use dense representation of graphs. I was struggled for a wihle for the same problem but can not figure out a getting around way except re-implementing a sparse version of the algorithm.

**Reply by rusty1s** [reply]
Yes, `DiffPool` is not really meant to scale to larger graphs and it is tricky or nearly impossible to implement a sparse version of it. Even in a sparse version of `DiffPool`, you would still need to learn a cluster assignment for every node, and your coarsened graphs become more and more dense by design.

There exists more sparsity-aware pooling approaches, e.g.,`TopKPooling`, although they are not as expressive as `DiffPool`.
